src/lib.rs

```diff
@@ -789,36 +611,35 @@ struct Runner<'a> {
     /// The executor state.
     state: &'a State,
 
-    /// Inner ticker.
-    ticker: Ticker<'a>,
-
     /// Bumped every time a runnable task is found.
     ticks: usize,
 }
 
 impl Runner<'_> {
     /// Creates a runner and registers it in the executor state.
     fn new(state: &State) -> Runner<'_> {
-        let runner = Runner {
-            state,
-            ticker: Ticker::new(state),
-            ticks: 0,
-        };
-        runner
+        state.local_queue.get_or_default().start_ticking();
+
+        Runner { state, ticks: 0 }
     }
 
     /// Waits for the next runnable task to run.
     async fn runnable(&mut self, rng: &mut fastrand::Rng) -> Runnable {
         let local = self.state.local_queue.get_or_default();
 
         let runnable = self
-            .ticker
-            .runnable_with(|| {
+            .state
+            .tick_with(|try_stealing| {
                 // Try the local queue.
                 if let Ok(r) = local.queue.pop() {
                     return Some(r);
                 }
 
+                // Remaining work involves stealing.
+                if !try_stealing {
+                    return None;
+                }
+
                 // Try stealing from the global queue.
                 if let Ok(r) = self.state.queue.pop() {
```

Comment:
here we're stealing from the global queue inside a closure given to `tick_with`, right?

Reply:
That is correct.